libraries/extractor/src/main/java/androidx/media3/extractor/mkv/MatroskaExtractor.java

```diff
@@ -744,6 +749,13 @@ protected void startMasterElement(int id, long contentPosition, long contentSize
           throw ParserException.createForMalformedContainer(
               "Multiple Segment elements not supported", /* cause= */ null);
         }
+
+        // If we have to reparse due to an IO exception we also have to clear the seek head data
```

Comment:
It's not obvious to me how you know that you've encountered an I/O exception if you land here

Is this the clearing logic I suggested maybe belongs in `MatroskaExtractor.seek()` in https://github.com/androidx/media/pull/2268#issuecomment-2762054677?